internal/factsengine/gatherers/fstab.go

```diff
@@ -60,7 +62,13 @@ func (f *FstabGatherer) Gather(ctx context.Context, factsRequests []entities.Fac
 
 	entries := []FstabEntry{}
 
-	for _, m := range mounts {
+	for i, m := range mounts {
+		if m.PassNo < 0 {
+			return nil, FstabFileDecodingError.Wrap("invalid check order for mount" + strconv.Itoa(i))
+		}
+		if m.Freq < 0 || m.Freq > math.MaxUint8 {
+			return nil, FstabFileDecodingError.Wrap("invalid backup frequency for mount" + strconv.Itoa(i))
+		}
```

Comment:
The change is needed to fix the lint error 
```
G115: integer overflow conversion int -> uint8
G115: integer overflow conversion int -> uint64
```

The _guards_ on the runtime value are needed to prove to the linter that both values are in the expected range.